projects/addon-editor/extensions/image-editor/image-editor.component.ts

```diff
@@ -56,14 +71,49 @@ export class TuiImageEditorComponent extends AbstractTuiEditorResizable<TuiEdita
         @Self()
         @Inject(TuiDestroyService)
         destroy$: TuiDestroyService,
+        @Inject(ElementRef) private readonly el: ElementRef<HTMLDivElement>,
+        @Inject(WINDOW) private readonly win: Window,
+        @Inject(ChangeDetectorRef) private readonly cd: ChangeDetectorRef,
+        @Inject(NgZone) private readonly ngZone: NgZone,
     ) {
         super(doc, destroy$);
     }
 
+    ngOnInit(): void {
+        fromEvent(this.editor.view.dom, 'click')
+            .pipe(
+                tuiZonefree(this.ngZone),
+                map(event => this.el.nativeElement.contains(event.target as Node | null)),
```

Comment:
Why simple `HostListener` for `click` wouldn't work?

Reply:
i don't know how to track the loss of focus outside `tui-image-editor` component

Reply:
`HostListener for click wouldn't work`
triggered only in `tui-img-editor` component area

`this.editor.view.dom === div[contenteditable=true].ProseMirror`

thus, we detect click event on the parent (this.editor.view.dom), and update focused state of the current `tui-img-editor` template through change detection

Reply:
Oh, I see, we also need to track `false`! Sorry, then your initial `@HostListener('document:click.silent', ['$event.target'])` was correct. For some reason I thought we only needed `true` values.

Reply:
fixed